components/butterfly/src/server/mod.rs

```diff
@@ -1335,6 +1347,123 @@ mod tests {
         assert_eq!(to_restart, vec![(service_group.to_string(), term)]);
     }
 
+    fn mock_service(member: &Member) -> Service {
+        Service {
+            member_id: member.id.clone(),
+            service_group: ServiceGroup::from_str("group.default").unwrap(),
+            incarnation: 0,
+            initialized: true,
+            pkg: "foo/bar".into(),
+            cfg: vec![],
+            sys: SysInfo::default(),
+        }
+    }
+
+    impl RumorStore<Service> {
+        fn contains(&self, service: &Service) -> bool {
```

Comment:
Would it potentially be worth adding this to the "real" `RumorStore<Service>`, rather than having it be just a test-only construct?

Reply:
Currently, I don't see any other places in the non-test code where it would be used, so I'd rather leave it in the test scope. That makes it clearer that it can be removed if the tests that use it no longer need it in the future.